### DISCUSSION TITLE
please make README.md reflect actual state of input-leap releases

### DISCUSSION THREAD

**Discussion by m10d** [input-leap/input-leap#1696] [Development of Input Leap]
### What happened?

![image](https://github.com/input-leap/input-leap/assets/59897595/190ff2dc-5d26-4ee3-be95-c7daac3e9def)

based on this, the project is ready to replace barrier.  Except there are no releases, and as best I can tell none available anywhere.  

Linux is trivial to build for, but windows (I don't use mac) requires a whole host of setup to build (and most of those same tools / vs libs to run the "released" bits that are available, which are only linked Debug.

Also, contrary to what's implied by #1414 (linked from the README.md), there appear to be breaking issues connecting `barrier` clients to an `input-leap` server.  
![image](https://github.com/input-leap/input-leap/assets/59897595/e7ead22e-5cac-4d66-a4f4-a24c5faf326e)

### Version

v2.0.0-RC1

### If applicable, where did you install Input Leap from?

from source on linux.

### What OSes are you seeing the problem on? (Check all that apply)

Windows

### What OS versions are you using?

host ubuntu20.04 desktop (x11)
windows10 tablet
(when wayland is supported, a ubuntu22.04 laptop)

**Comment by shymega** [comment]
Thank you for bringing this up.

I've made a draft PR #1697 and would be keen for the community's thoughts.

Thank you.

**Comment by decryptedchaos** [comment]
It's a fair point,  i've used everything since Synergy first came out,   Barrier dropped the ball kinda,   and while this project seems to be improving on the progress from barrier this one issue is glaring..  

_I Appoligize if this sounds like a rant_ This project,  forked in 2021,  and its a small project and a small team,  so that gives it a lot of leeway in understanding that things take time..   However,  release are incredibly hard to locate,  it took me until today to find them in Azure,  this is highly highy  cumbersome to the end users,  azure isn't intergrated well with github and going to the actions page does not link to the artifact,  it lists it,  but clicking it does NOTHING..  

In my opinion using Azure for CI is hurting this project,  especially for windows users,   if this were a random Linux app it wouldn't be as big of a deal we have build scripts to take care of this for us on Linux,  but windows is a whole other animal,  its not so straight forward to compile something in windows as it is in linux  the tooling and dependancys are often more trouble than its worth.. that a end user,  even a power users like myself will just consider other options before dealing with it..  

So yeah,  i assume there is a reason for using Azure, i suspect MS has given some incentive to use it,  or whatever. the case,   but it needs better intergration and/or configuration to more easily present the build artifacts on github  it's incredibly difficult to find them.

**Reply by TheAkashicTraveller** [reply]
I agree but so far as the average user is concerned this doesn't even exist. There still isn't a single release when they should have made one week 1 even if it was just barrier without even all the name references changed. Nevermind years later. This perfectionism of having a major release as the first one is a really bad decisions imo.

**Reply by shymega** [reply]
We are volunteers. We don't get paid for this, and the route we're taking is our decision. There are remaining issues that need to be solved for a major release, so PRs that _resolve_ this issue are welcome.

However, telling us how we should manage the project, when we do this voluntarily, doesn't motivate me, at least.

Azure-wise, that will soon be removed, and we'll be using GH Actions. But I'm waiting for that PR to be approved and merged. I can't speed this along. Also, I'd just like to say that the idea of Microsoft giving us an incentive to use Azure Pipelines is ludicrous. Please, think twice before speculating wild things like this.

A release will come when it's ready. If you want to step up and help with PRs, and remaining bugs (some have been added to the v3 milestone), then, please, do so.

I - @shymega - am trying my best to keep this project afloat. But I'm finding myself feeling burnt out, and I just want to reiterate the following: open-source is a _privilege_, not a right.

**Comment by decryptedchaos** [comment]
>However, telling us how we should manage the project, when we do this voluntarily, doesn't motivate me, at least.

That tired argument that we always get from developers like you  Oh we're volunteers omg pity us,  it gets old real fast when its used as a means to belittle very legitimate criticism, Now I gave you the leeway in my original post,  I acknowledged that open source projects do take time,  but you, using this as an excuse after 3 years is abysmal at best..  there is in fact zero reason for at least one release not to exist on GitHub I don't care if there wasn't a single bug fix in it..  there's no reason for it not to exist..   You've had three years to do it, I don't even want to hear the word "time"   

See, I'm a forgiving person,  I point you back to my acknowledgement that part time volunteer projects do take time, but using this as an excuse after so long is questionable, but more than that,  using it to discredit users constructive criticisms  say's way more about you than it does me.  because while you may see it as unimportant,  we're here telling you that isn't the case..  and I am sure many other users met with the same fate and just didn't bother to say anything just assumed it was a dead project..  And you say that Azure is going away for GH Actions okay fine,  what looks out of place is why has it been in this fracture state like this for 3 years.  why hasn't someone on the team, yourself included just cherrypicked one build and said there that one   lets make that our inaugural release until we get everything the way we want it if you're so hell bent on perfectionism.

So dismissing our concern as you have with that little display,  tells me everything I need to know about why this project is the way it is,  for nearly three years,  clearly you're a pedantic ass,  Telling you how to run the project huh?  I'm telling you you're current structure of the project is stupid and making it very difficult for people to find the binary's.  but you keep telling yourself that..   You're getting the third degree now  because you are rude AF  when both of us are "CONTRIBUTING" to the project by offering our insight and advice  based on our experience with the project and its current state..   But you and people like you don't want to acknowledge that,  all you want to talk about is  fix what I want, when I want, when I say or shut up.  And definitely don't question me for any reasons. 

Azure sticks out like a sore thumb on GitHub,  its reasonable to question its reason for being used, and I don't think it is to much of a stretch to speculate that MS offers incentives to use it.

>I - @shymega - am trying my best to keep this project afloat. But I'm finding myself feeling burnt out

It's not that your time isn't appreciated,  it's that you forget what the actual point is,  its to be used,  by people, many of which are probably having trouble even finding/installing it much less using it..  You say you want to keep it afloat,  well then I would say belittling the users you have that were willing to take the time to point out a area that needs improvements is a poor way to do that

From my perspective this project is failing its mission,  it forked from barrier because of mismanagement,  but is too mismanaged itself,  if you can't at least try to do the normal user onboarding stuff most projects do like have easily accessible builds..  

I have hereby forked this project  in three different locations,  because maybe it needs yet another management,  with that said I have too many projects going so I won't be rushing to compete with this project anytime soon,  but if this goes stagnate like projects of the past have shown I will be there to give it its third life,  and I'll actually release binary's on the GitHub page in fact I'll go to my fork tonight and upload the latest build to it.. to prove its not that hard..  

https://github.com/decryptedchaos/input-leap/releases

>and I just want to reiterate the following: open-source is a privilege, not a right.

Are you crazy?  that is in fact the definition of "Open Source"  it was created out of the "Belief" that users should have the "RIGHT" to free open software.  and not be beholden to large corporations.  It certainly wasn't a belief in a classist privilege system..  that idea in of it self goes so against the very spirit of open source in the first place..  


What brought this attitude to you was your own,  I was perfectly polite in my original post..  But you were nothing but aggressive and belligerent from the first sentence to the last..  I would say that your need to be so defensive says more about you than you think..  but  I digress..  I've said what I need to say..   

I hope things improve here,  the idea behind the project is a good one, bus this attitude in open source is very much hurting open source..  you can't tell the contributions that you don't want to hear from them because they aren't contributing what you want them to contribute.. 
so what if we either don't have the time or skill to fix every single bug in the code base before you say its good enough to release a binary,  we're still contributing by advice..   and your treatment of that advice has left me with a bad taste...   Indeed...

**Reply by abienz** [reply]
I think you need to take a break @decryptedchaos

**Reply by ksandom** [reply]
> So dismissing our concern as you have with that little display, tells me everything I need to know about why this project is the way it is, for nearly three years, clearly you're a pedantic ass, 

Exactly where did @shymega do that?

Here's the thing. You do not, and should not, have visibility into how much time open source developers are able to put into any of their projects. You do not, and should not, have visibility into competing aspects of their life. So if something isn't up to your standards, you have exactly 3 ways you can help:

* Raise a Pull Request that solves the problem.
* Create an issue that outlines what needs to be fixed. Include solutions as well a the problem. It will then be prioritised against the other tasks and acted on, or not, according to where it sits with the competing priorities.
  * \[You don't have to agree with those priorities, and you can give your opinion. But you do have to accept them.\] 
* Finance the development of the thing that concerns you.

Berating them for not having the same priorities as you, however, is not an option.

I stopped reading pretty early, so if you do have something constructive to say, please say it again in a more concise, polite form, and I'd love to hear it. 

It sounds like this project is important to you. Please show that by helping out.

**Comment by decryptedchaos** [comment]
>It sounds like this project is important to you. Please show that by helping out.

As I have already said,  I am helping out,  just because its  "Not up to you're standards"  doesn't make it any less so...  see we can both play that game..  

The argument all of you are make is stupid plain and simple..  we haven't released a single binary on github were every single user expect to find them in 3 years because "uh bugs exist.."  that has to be the dumbest logic I've heard in a while..  

Look there is zero defense to not having a binary available on github from the first day of the fork  plain and simple..  

Taking the stance of "Uhhh well finance our project bro if you want a binary"  is unacceptable.. period

Any kind of argument against why a single binary hasn't been published to github in three years is a pointless argument there is zero justification for it..  

You act like its hard to do,  I published a binary on my fork in 30 seconds..  and you want me to believe that the 35 of you couldn't take 30 second to publish a binary to github?   really??   

That right there is justification for my attitude,  you want to gaslight us into believing that it would take you so much time to upload a 9MB file..  you know it wouldn't and yet you persist with this rhetoric..  its radicicolous as far as I'm concerned,  and I will not stand by and abide the insult to my intelligence 
 
And that's the last thing I'm going to say, because I'm going to bed.

**Comment by shymega** [comment]
I'm locking this. Windows builds are now publishing, but I need to fix macOS builds. There are also RPMs being built.

At the end of the day, abuse won't be tolerated. Remember that when speaking to a fellow human being.
